.editorconfig

```diff
@@ -0,0 +1,21 @@
+# http://EditorConfig.org
+
+root = true
+
+# Defaults apply to all files
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```

Comment:
This one I'm not sure about: obviously we don't want trailing whitespace, but we also don't want commits to include trimming of trailing whitespace in unrelated code. Such trailing whitespace might be introduced accidentally by users with editors not honoring .editconfig.

What do you think is best?

Reply:
I have `trim_trailing_whitespace = true` for all my work projects, and haven't had any problems — but I'm mostly working with people using IDEs or editors that have built-in support for Editorconfig.

One other option is a Git pre-commit hook, but this [looks complicated](https://github.com/imoldman/config/blob/master/pre-commit.git.sh) so I won't recommend it.

Reply:
Another option would be a travis check that fails if people add trailing whitespace.

Reply:
I prefer Travis. Trimming is turned on by default in Atom for example and it caused some PRs to be cluttered with such small fixes not related to the subject.

Reply:
If Travis were set to fail because of trailing whitespace, then adding the same rule in Editorconfig is convenient for users whose editors support it.

At present, the only files with trailing whitespace are Markdown — they could be excluded from this rule.